numpy/core/_add_newdocs.py

```diff
@@ -7970,62 +7972,164 @@ def luf(lamdaexpr, *args, **kwargs):
-add_newdoc('numpy.core.numerictypes', 'float32',
+add_newdoc_for_numeric_type('ushort', [], unsigned_integer_aliases,
     """
-    32-bit floating-point number. Character code 'f'. C float compatible.
+    Unsigned integer type, compatible with C ``unsigned short``.
+    """)
 
+add_newdoc_for_numeric_type('uintc', [], unsigned_integer_aliases,
+    """
+    Unsigned integer type, compatible with C ``unsigned int``.
     """)
 
-add_newdoc('numpy.core.numerictypes', 'float64',
+add_newdoc_for_numeric_type('uint', [], unsigned_integer_aliases,
     """
-    64-bit floating-point number. Character code 'd'. Python float compatible.
+    Unsigned integer type, compatible with C ``unsigned long``.
+    """)
 
+add_newdoc_for_numeric_type('ulonglong', [], unsigned_integer_aliases,
+    """
+    Signed integer type, compatible with C ``unsigned long long``.
     """)
 
-add_newdoc('numpy.core.numerictypes', 'float96',
+add_newdoc_for_numeric_type('half', [], float_aliases,
     """
+    Half-precision floating-point number type.
     """)
 
-add_newdoc('numpy.core.numerictypes', 'float128',
+add_newdoc_for_numeric_type('single', [], float_aliases,
     """
-    128-bit floating-point number. Character code: 'g'. C long float
-    compatible.
+    Single-precision floating-point number type, compatible with C ``float``.
+    """)
 
+add_newdoc_for_numeric_type('double', ['float_'], float_aliases,
+    """
+    Double-precision floating-point number type, compatible with Python `float`
+    and C ``double``.
     """)
 
-add_newdoc('numpy.core.numerictypes', 'int8',
-    """8-bit integer. Character code ``b``. C char compatible.""")
+add_newdoc_for_numeric_type('longdouble', ['longfloat'], float_aliases,
+    """
+    Extended-precision floating-point number type, compatible with C
+    ``long double`` but not necessarily with IEEE 754 quadruple-precision.
+    """)
 
-add_newdoc('numpy.core.numerictypes', 'int16',
-    """16-bit integer. Character code ``h``. C short compatible.""")
+add_newdoc_for_numeric_type('csingle', ['singlecomplex'], float_aliases,
+    """
+    Complex number type composed of two single-precision floating-point
+    numbers.
+    """)
 
-add_newdoc('numpy.core.numerictypes', 'int32',
-    """32-bit integer. Character code 'i'. C int compatible.""")
+add_newdoc_for_numeric_type('cdouble', ['cfloat', 'complex_'], float_aliases,
+    """
+    Complex number type composed of two double-precision floating-point
+    numbers, compatible with Python `complex`.
+    """)
 
-add_newdoc('numpy.core.numerictypes', 'int64',
-    """64-bit integer. Character code 'l'. Python int compatible.""")
+add_newdoc_for_numeric_type('clongdouble', ['clongfloat', 'longcomplex'], float_aliases,
+    """
+    Complex number type composed of two extended-precision floating-point
+    numbers.
+    """)
 
-add_newdoc('numpy.core.numerictypes', 'object_',
-    """Any Python object.  Character code: 'O'.""")
+add_newdoc_for_numeric_type('object_', [], [],
```

Comment:
Not sure I'd consider this numeric. Probably better either to rename the function to `*_scalar_type`, or leave the non-numeric types as they were.